restapi/src/test/java/io/stargate/db/datastore/ValidatingDataStore.java

```diff
@@ -96,22 +96,36 @@ public boolean supportsLoggedBatches() {
         .findAny()
         .orElseGet(() -> Assertions.fail("Unexpected query: " + queryString));
 
-    Prepared<B> p = new Prepared<>(query);
-    prepared.add(p);
-    return CompletableFuture.completedFuture(p);
+    // Generate unique IDs for each `prepare` call to validate that all prepared statements get
+    // executed.
+    ByteBuffer buffer = ByteBuffer.allocate(Integer.BYTES);
+    buffer.putInt(preparedIdSeq.incrementAndGet());
+    MD5Digest preparedId = MD5Digest.wrap(buffer.array());
+    Query<B> queryWithId = query.withPreparedId(preparedId);
+    prepared.computeIfAbsent(preparedId, p -> new Prepared<>(queryWithId));
+    return CompletableFuture.completedFuture(queryWithId);
```

Comment:
nit: maybe extract this logic to a separate method with a name stating the intent clearly. (emphasize more what's in the comment)

Reply:
refactored (force-pushed)